worker-document/src/main/java/com/hpe/caf/worker/document/impl/ScriptImpl.java

```diff
@@ -237,6 +263,31 @@ public void raiseEvent(final String event, final Object... args)
 
         // Check if there is a JavaScript function event handler for the event
         final Object eventHandler = bindings.get(event);
+        if (scriptSpec.getEngineType() == ScriptEngineType.GRAAL_JS) {
+            graalHandleEvent(eventHandler, args);
+        } else {
+            nashornHandleEvent(eventHandler, args);
+        }
+    }
+
+    private static void graalHandleEvent(final Object eventHandler, final Object[] args)
+    {
+        if (!(eventHandler instanceof Function)) {
```

Comment:
Is this check really required?  `Function` normally only takes one argument.  Does this just happen to work because our JS event handlers all take one argument?

Reply:
The docs aren't super clear on this, but a guest language function (or 'executable' in their terminology) will implement `Function<Object[], Object>` while and object is returned as a `Map` and primitives are just mapped directly to the Java type.